### ISSUE TITLE
[0 (WaitingForEndOfRequestEntity)] Slot execution failed.

### ISSUE THREAD

**Issue by ajmnsk** [akka/akka-http#2079]
Our Akka-Http = **"10.1.1"**, and Akka = **"2.5.13"**
We have following error. **akka-http** application is acting as a client, and executing REST calls to GET, and POST data:

```
2018-06-18 13:55:23.745UTC ERROR a.h.impl.engine.client.PoolGateway LmcAkkaHttpServer/Pool(shared->http://services-server:8080) - [0 (WaitingForEndOfRequestEntity)] Slot execution failed. That's probably a bug. Please file a bug at https://github.com/akka/akka-http/issues. Slot is restarted.
java.lang.IllegalStateException: Cannot [onResponseReceived] when in state [WaitingForEndOfRequestEntity]
	at akka.http.impl.engine.client.pool.SlotState.illegalState(SlotState.scala:84)
	at akka.http.impl.engine.client.pool.SlotState.onResponseReceived(SlotState.scala:61)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$Event$.$anonfun$onResponseReceived$1(NewHostConnectionPool.scala:130)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$Slot.runOneTransition$1(NewHostConnectionPool.scala:208)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$Slot.loop$1(NewHostConnectionPool.scala:282)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$Slot.updateState(NewHostConnectionPool.scala:291)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$Slot.onResponseReceived(NewHostConnectionPool.scala:182)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$SlotConnection.$anonfun$onPush$8(NewHostConnectionPool.scala:424)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$SlotConnection.$anonfun$onPush$8$adapted(NewHostConnectionPool.scala:424)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$SlotConnection.withSlot(NewHostConnectionPool.scala:371)
	at akka.http.impl.engine.client.pool.NewHostConnectionPool$HostConnectionPoolStage$$anon$1$SlotConnection.onPush(NewHostConnectionPool.scala:424)
	at akka.stream.stage.GraphStageLogic$SubSinkInlet.$anonfun$_sink$1(GraphStage.scala:1282)
	at akka.stream.stage.GraphStageLogic$SubSinkInlet.$anonfun$_sink$1$adapted(GraphStage.scala:1277)
	at akka.stream.impl.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:454)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:468)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:563)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:745)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:735)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:761)
	at akka.actor.Actor.aroundReceive(Actor.scala:517)
	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:670)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
	at akka.actor.ActorCell.invoke(ActorCell.scala:557)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
2018-06-18 13:55:23.745UTC WARN  a.h.impl.engine.client.PoolGateway LmcAkkaHttpServer/Pool(shared->http://services-server:8080) - [0 (WaitingForEndOfRequestEntity)] Ongoing request [POST /app/v5/documents Chunked] was dropped because pool is shutting down
```

**Comment by johanandren** [comment]
Can you repeat the error or was it a one-off random error?

**Comment by ajmnsk** [comment]
It started happening after some time. Application was up the whole week, and after the week-end (no activity from client side), issue started, and, it looked like, all subsequent requests were failing with the same error until akka-http application was re-started. Also, we have a way to bring the application down from browser by executing following, and this request was failing as well, meaning, it was not going down, just log.info("System shutdown request received.") was getting reported .. ?

        path("shutdown") {
          import scala.concurrent.ExecutionContext.Implicits.global
          val r: Future[Unit] = Http().shutdownAllConnectionPools().andThen { case _ => system.terminate() }
          log.info("System shutdown request received.")
          var code: Int = 200
          var msg: String = "System shutdown successfully."
          r.onComplete {
            case Success(_) =>
              log.info(msg)
            case Failure(e) =>
              val msg = "Error occurred during system shutdown: " + e.getMessage
              log.error(msg)
              code = 500
          }
          complete((code, msg))
        }

**Comment by raboof** [comment]
I suspect this might have been fixed by #1743 , could you try with akka-http 10.1.3?

**Comment by ajmnsk** [comment]
We'll upgrade to akka-http 10.1.3, and test. Thanks.

**Comment by raboof** [comment]
(closing on the theory that this was indeed fixed in 10.1.3, feel free to reopen when you still see it with a later version)
